### PR TITLE
823 too many expanduser

### LINKED ISSUE #823: Reduce the number of "expanduser" filters to a minimum
There are over 200 lines with the exact same `expanduser` usage. Reduce these to the minimum number required to produce the same results.

### PR DESCRIPTION
**Pull request by joejulian**
By ensuring that `config_base` and `config_file_default` are expanded early, this merge request mostly eliminates the use of the expanduser filter making the code much easier to read.

Fixes #823

### CODE REVIEW

File: ansible/roles/kraken.config/defaults/main.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 kraken_action: up
-config_base: "~/.kraken"
-config_file_default: "{{ config_base | expanduser }}/config.yaml"
+config_base: '{{ "~/.kraken" | expanduser }}'
```
Comment (davidewatson) [inline ansible/roles/kraken.config/defaults/main.yaml]: What about the case where `config_base` is passed into Ansible from the `./up.sh` or `kraken`? I think you need to use the `expanduser` filter from a task list and not the defaults, maybe in `kraken.config` + `/tasks/global-setup.yaml`?

**Comment by mikeln** [comment]
e2e tests do not appear to relate to the change made.  should be ok to merge.

File: ansible/roles/kraken.config/defaults/main.yaml (hunk 1 above)
Comment (joejulian) [reply ansible/roles/kraken.config/defaults/main.yaml]: I handled that in `lib/common.sh` by expanding the path passed in. Do you think that should _also_ be done in ansible for clarity?

Comment (davidewatson) [reply ansible/roles/kraken.config/defaults/main.yaml]: Our defaults are scattered and only partially documented. For example, does the default come from `./lib/common.sh`, ansible defaults (e.g. `kraken.*/defaults/main.yaml`), an ansible task list, or jsonschema? I'd prefer if they were all set in one place. I think that means an ansible task list, preferably in `kraken.config`. It's your call.

Interestingly, `realpath` doesn't actually expand tilde:

```
bash-4.3# realpath ~/
/root
bash-4.3# realpath '~/'
realpath: ~/: No such file or directory
bash-4.3# 
```

So your current code is depending on bash.

Comment (joejulian) [reply ansible/roles/kraken.config/defaults/main.yaml]: Huh, interesting. It worked in Linux.

I was already planning on removing that and putting the expansion, once, in tasks instead but I'm waiting to rebase it after #826 is merged.

**Pull request merged**